pkg/types/spec/validations.go

```diff
@@ -1150,6 +1187,58 @@ func validateONNXPredictor(api *userconfig.API, models *[]CuratedModelResource,
 	return nil
 }
 
+func validateONNXModelFilePath(modelPath string, projectDir string, awsClient *aws.Client, gcpClient *gcp.Client) error {
+	s3Path := strings.HasPrefix(modelPath, "s3://")
+	gcsPath := strings.HasPrefix(modelPath, "gs://")
+	localPath := !s3Path && !gcsPath
+
+	if s3Path {
+		awsClientForBucket, err := aws.NewFromClientS3Path(modelPath, awsClient)
+		if err != nil {
+			return err
+		}
+
+		bucket, modelPrefix, err := aws.SplitS3Path(modelPath)
+		if err != nil {
+			return err
+		}
+
+		yes, err := awsClientForBucket.IsS3Prefix(bucket, modelPrefix)
```

Comment:
nit: consider renaming `yes` to `isS3Prefix` (also below)

Reply:
Makes sense.